barretenberg/ts/src/barretenberg/verifier.ts

```diff
@@ -13,17 +13,22 @@ export class BarretenbergVerifier {
   // eslint-disable-next-line  @typescript-eslint/no-explicit-any
   private acirComposer: any;
 
-  constructor(private options: BackendOptions = { threads: 1 }) {}
+  constructor(private options: BackendOptions = { threads: 1 }, api?: Barretenberg) {
+    // For sharing if also initializing backend
+    if (api) {
+      this.api = api;
+    }
+  }
```

Comment:
We should probably rework this so the constructor just accepts a `Barretenberg` and have a `Verifier::new` method if we want to init the wasm at the same time.